source/server/http/admin.cc

```diff
@@ -594,10 +596,15 @@ Http::Code AdminImpl::handlerPrometheusStats(absl::string_view, Http::HeaderMap&
 }
 
 std::string PrometheusStatsFormatter::sanitizeName(const std::string& name) {
-  std::string stats_name = name;
-  std::replace(stats_name.begin(), stats_name.end(), '.', '_');
-  std::replace(stats_name.begin(), stats_name.end(), '-', '_');
-  return stats_name;
+  // The name must match the regex [a-zA-Z_:][a-zA-Z0-9_:]* as required by
+  // prometheus. Of which the colon is reserved and should not be exported.
+  // Refer to https://prometheus.io/docs/concepts/data_model/.
```

Comment:
Mild preference that we just update the regex.

  // The name must match the regex [a-zA-Z_][a-zA-Z0-9_]* as required by prometheus. 
  // Refer to ...

Reply:
Updated.